q2_vsearch/tests/test_cluster_features.py

```diff
@@ -508,34 +438,191 @@ def test_no_overlapping_feature_ids(self):
+
+class ClusterFeaturesOpenReference(TestPluginBase):
 
     package = 'q2_vsearch.tests'
 
-    def test_fasta_with_sizes(self):
+    def setUp(self):
+        super().setUp()
+        self.open_reference = \
+            self.plugin.pipelines['cluster_features_open_reference']
+
         input_sequences_fp = self.get_data_path('dna-sequences-1.fasta')
+        self.input_sequences = Artifact.import_data('FeatureData[Sequence]',
+                                                    input_sequences_fp)
+
+        ref_sequences_fp = self.get_data_path('ref-sequences-1.fasta')
+        self.ref_sequences = Artifact.import_data('FeatureData[Sequence]',
+                                                  ref_sequences_fp)
+
         input_table = biom.Table(np.array([[100, 101, 103],
                                            [1, 1, 2],
                                            [4, 5, 6],
                                            [7, 8, 9]]),
                                  ['feature1', 'feature2', 'feature3',
                                   'feature4'],
                                  ['sample1', 'sample2', 'sample3'])
+        self.input_table = Artifact.import_data('FeatureTable[Frequency]',
+                                                input_table)
+        self.exp_seqs = _read_seqs(self.input_sequences)
+
+    def test_100_percent_clustering(self):
+        exp_table = biom.Table(np.array([[100, 101, 103],
+                                         [1, 1, 2],
+                                         [4, 5, 6],
+                                         [7, 8, 9]]),
+                               ['r1', 'feature2', 'feature3', 'r2'],
+                               ['sample1', 'sample2', 'sample3'])
+
+        with redirected_stdio(stderr=os.devnull):
+            obs_table, rep_seqs, new_ref_seqs = self.open_reference(
+                sequences=self.input_sequences, table=self.input_table,
+                reference_sequences=self.ref_sequences, perc_identity=1.0)
+
+        obs_table = obs_table.view(biom.Table)
+        # order of identifiers is important for biom.Table equality
+        obs_table = \
+            obs_table.sort_order(exp_table.ids(axis='observation'),
+                                 axis='observation')
+        self.assertEqual(obs_table, exp_table)
+
+        obs_rep_seqs = _read_seqs(rep_seqs)
+        exp_rep_seqs = [self.exp_seqs[1], self.exp_seqs[2], self.exp_seqs[0],
+                        self.exp_seqs[3]]
+        exp_rep_seqs[2].metadata['id'] = 'r1'
+        exp_rep_seqs[3].metadata['id'] = 'r2'
+        self.assertEqual(obs_rep_seqs, exp_rep_seqs)
+
+        obs_ref_seqs = _read_seqs(new_ref_seqs)
+        ref_seqs = _read_seqs(self.ref_sequences)
+        exp_ref_seqs = [self.exp_seqs[1], self.exp_seqs[2], ref_seqs[0],
+                        ref_seqs[1]]
+        self.assertEqual(obs_ref_seqs, exp_ref_seqs)
+
+    def test_97_percent_clustering(self):
```

Comment:
This test, paired with `test_100_percent_clustering`, shows that `perc_identity` is impacting closed-reference clustering. Could you add a `feature5` that clusters with `feature2` at 97% identity, but not at 100% identity, so we can see that `perc_identity` is also impacting de novo clustering?